lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -852,19 +852,12 @@ class ExprEmitter : public EmitterBase,
 SubExprInfo ExprEmitter::emitBinary(Operation *op, VerilogPrecedence prec,
                                     const char *syntax,
                                     SubExprSignRequirement operandSignReq) {
+  auto lhsPrec = VerilogPrecedence(prec - 1);
   auto lhsInfo =
-      emitSubExpr(op->getOperand(0), prec, OOLBinary, operandSignReq);
+      emitSubExpr(op->getOperand(0), lhsPrec, OOLBinary, operandSignReq);
   os << ' ' << syntax << ' ';
 
-  // The precedence of the RHS operand must be tighter than this operator if
-  // they have a different opcode in order to handle things like "x-(a+b)".
-  // This isn't needed on the LHS, because the relevant Verilog operators are
-  // left-associative.
-  //
   auto rhsPrec = VerilogPrecedence(prec - 1);
```

Comment:
nice great catch.  Please pull both of these up into a "subexprPrec".  Also, why are you dropping this logic:

```
  if (auto *rhsOperandOp = op->getOperand(1).getDefiningOp())
    if (op->getName() == rhsOperandOp->getName())
      rhsPrec = prec;
```

?  I think the code should be correct with this due to the associativity, and this drops extraneous parens.

Reply:
This is specifically the incorrect code. == is not associative.